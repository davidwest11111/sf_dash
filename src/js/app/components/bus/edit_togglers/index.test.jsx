import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import LocalStorage from 'app/utils/local_storage';

import EditTogglers from './index';


describe('outputs the expected tree when', () => {
  let wrapper;
  let defaultProps;

  beforeEach(() => {
    defaultProps = {
      showBusRemove: false,
      toggleShowBusRemove: () => {},
      toggleAddStopForm: () => {},
    };
  });

  test('both buttons are shown', () => {
    jest.spyOn(LocalStorage, 'getBusStopsFromLocalStorage')
      .mockReturnValue(['14|1234']);
    wrapper = shallow((
      <EditTogglers
        {...defaultProps}
        showBusRemove={false}
      />
    ));
  });

  test('only the add button is shown', () => {
    jest.spyOn(LocalStorage, 'getBusStopsFromLocalStorage')
      .mockReturnValue([]);
    wrapper = shallow((
      <EditTogglers
        {...defaultProps}
        predictions={[]}
        showBusRemove={false}
      />
    ));
  });

  test('only the edit button is shown', () => {
    jest.spyOn(LocalStorage, 'getBusStopsFromLocalStorage')
      .mockReturnValue(['14|1234']);
    wrapper = shallow((
      <EditTogglers
        {...defaultProps}
        showBusRemove
      />
    ));
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
