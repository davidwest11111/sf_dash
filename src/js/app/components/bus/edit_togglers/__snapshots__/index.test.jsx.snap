// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when both buttons are shown 1`] = `
<div
  className=""
>
  <Button
    className="mr-1"
    color="secondary"
    onClick={[Function]}
    outline={true}
    size="xs"
    tag="button"
  >
    <span
      className="fas fa-plus"
    />
  </Button>
  <Button
    className="mr-1"
    color="secondary"
    onClick={[Function]}
    outline={true}
    size="xs"
    tag="button"
  >
    <span
      className="fas fa-pen"
    />
  </Button>
</div>
`;

exports[`outputs the expected tree when only the add button is shown 1`] = `
<div
  className=""
>
  <Button
    className="mr-1"
    color="secondary"
    onClick={[Function]}
    outline={true}
    size="xs"
    tag="button"
  >
    <span
      className="fas fa-plus"
    />
  </Button>
</div>
`;

exports[`outputs the expected tree when only the edit button is shown 1`] = `
<div
  className=""
>
  <Button
    className="mr-1"
    color="secondary"
    onClick={[Function]}
    outline={true}
    size="xs"
    tag="button"
  >
    <span
      className="fas fa-pen"
    />
  </Button>
</div>
`;
